fiftyone/core/collections.py

```diff
@@ -69,49 +71,97 @@ def registrar(func):
 aggregation = _make_registrar()
 
 
-class _AutosaveSample:
-    def __init__(self, batch, sample):
-        self._batch = batch
-        self._sample = sample
+class SaveContext(object):
+    """Context that saves samples from a collection according to a configurable
+    batching strategy.
+
+    Args:
+        sample_collection: a
+            :class:`fiftyone.core.collections.SampleCollection`
+        batch_size (None): the batching strategy to use. Can either be an
+            integer specifying the number of samples to save in a batch, or a
+            float number of seconds between batched saves
+    """
+
+    def __init__(self, sample_collection, batch_size=None):
+        if batch_size is None:
+            batch_size = 0.2
```

Comment:
Time-based seems like a reasonable default to me because it controls the DB connection rate (soon, the API request rate), which is something one often wants to limit. 

A fixed batch size would produce wildly different connection rates since the size of `Sample` objects may vary from tiny to 16MB